compiler/ast.nim

```diff
@@ -851,6 +851,8 @@ type
   TSym* {.acyclic.} = object of TIdObj # Keep in sync with PackedSym
     # proc and type instantiations are cached in the generic symbol
     case kind*: TSymKind
+    of skModule:
+      realModule*: PSym # for `createModuleAlias`
```

Comment:
Does this need IC support?

Reply:
done, but untestable beyond the existing ic tests (that still do pass) because of pre-existing issue with ic+export: https://github.com/nim-lang/Nim/issues/17943; new tests can be added in future work.

Also, manually serializing/deserializing the AST/symbols/various compiler structures as in done in packed_ast is IMO the wrong approach, it duplicates logic and adds significant complexity, and on top of that in my experience using ic makes compilation slower that a full recompilation. We should really explore alternatives:
* using a generic serializer/deserializer that works with any types including loopy structures, plus ability to customize in the rare places where it needs to; it's not hard and much more maintaniable
* the compiler server approach we discussed before, which IMO is the best (still need to share my code, what worked, and what didn't yet work)

Reply:
Well since you bring it up:

> using a generic serializer/deserializer that works with any types including loopy structures, plus ability to customize in the rare places where it needs to; it's not hard and much more maintaniable

Well if IC is already too slow, a more generic solution won't make it faster... Furthermore I regard the .rod file format as the future for Nim's tooling and it enforces us to be honest about phase ordering issues (no more unprincipled mutations possible). A compiler server accomplishes nothing comparable. We already have nimsuggest, it eats gigabytes of RAM and occasionally 100% CPU...

The "loopy structures" are harmful and cause many many bugs; don't write tooling to hide them, avoid them.